src/app/modules/user/user.utils.ts

```diff
@@ -0,0 +1,57 @@
+import { StatusCodes } from 'http-status-codes'
+
+import { createLoggerWithLabel } from '../../../config/logger'
+import { SmsSendError } from '../../services/sms.service'
+import * as CoreErrors from '../core/core.errors'
+import { ErrorResponseData } from '../core/core.types'
+
+import * as UserErrors from './user.errors'
+
+const logger = createLoggerWithLabel(module)
+/**
+ * Handler to map ApplicationErrors to their correct status code and error
+ * messages.
+ * @param error The error to retrieve the status codes and error messages
+ * @param coreErrorMessage Any error message to return instead of the default core error message, if any
+ */
+export const mapRouteError = (
+  error: CoreErrors.ApplicationError,
+  coreErrorMessage?: string,
+): ErrorResponseData => {
+  switch (error.constructor) {
+    case UserErrors.InvalidOtpError:
+      return {
+        statusCode: StatusCodes.UNPROCESSABLE_ENTITY,
```

Comment:
why is this a 422? this is invalid authentication, so should be 401 right?

Reply:
Makes sense; I was following the previous implementation, but I agree 401 is the better code to return.

Reply:
need to change the alarms again though lol